### PR TITLE
Output HostStatsCollector's Collect errors

### PR DESCRIPTION
**Pull request by nak3**

### CODE REVIEW

File: client/stats/host.go [hunk 1]
```diff
@@ -80,6 +83,8 @@ func (h *HostStatsCollector) Collect() (*HostStats, error) {
 			Free:      memStats.Free,
 		}
 		hs.Memory = ms
+	} else {
```
Comment (diptanu) [inline client/stats/host.go]: Let's remove the `if/else` statements.

```
memStats, err := mem.VirtualMemory()
....
if err != nil {
   mErr.Errors = append(mErr.Errors, fmt.Errorf("error to correct memory stats", err))
}
```

**Comment by diptanu** [comment]
I initially didn't want to report any errors since I wanted to collect as much stats I could in the event stats collection of a particular sub-system was failing.

File: client/stats/host.go (hunk 1 above)
Comment (nak3) [inline client/stats/host.go]: I agree that removing `if/else` statements. But `return nil, err` is OK?

~~~
if err != nil {
   return nil, err
}
~~~

I thought it should collect errors as below like you commented.

~~~
mErr.Errors = append(mErr.Errors, fmt.Errorf("error to collect cpu stats", err))
~~~

Comment (diptanu) [inline client/stats/host.go]: @nak3 Yeah this is ok, since even if we collect more stats but return an error we won't update the stats.

Comment (nak3) [inline client/stats/host.go]: @diptanu I see. Thank you.

File: command/agent/config_test.go [hunk 2]
```diff
@@ -139,7 +139,7 @@ func TestConfig_Merge(t *testing.T) {
 				"foo": "bar",
 				"baz": "zip",
 			},
-			ChrootEnv: map[string]string{},
+			ChrootEnv:      map[string]string{},
```
Comment (nak3) [inline command/agent/config_test.go]: Here is a fix for gofmt error.

**Comment by nak3** [comment]
@diptanu Thank you. I updated.

File: client/stats/host.go [hunk 3]
```diff
@@ -72,67 +72,78 @@ func NewHostStatsCollector() *HostStatsCollector {
...
-			Available: memStats.Available,
-			Used:      memStats.Used,
-			Free:      memStats.Free,
-		}
-		hs.Memory = ms
+	memStats, err := mem.VirtualMemory()
+	if err != nil {
+		return nil, err
+	}
+	ms := &MemoryStats{
```
Comment (diptanu) [inline client/stats/host.go]: `hs.Memory := &MemoryStats {`

File: client/stats/host.go [hunk 4]
```diff
@@ -72,67 +72,78 @@ func NewHostStatsCollector() *HostStatsCollector {
...
+		Total:     memStats.Total,
+		Available: memStats.Available,
+		Used:      memStats.Used,
+		Free:      memStats.Free,
 	}
+	hs.Memory = ms
```
Comment (diptanu) [inline client/stats/host.go]: Do this inline above

File: command/agent/config_test.go (hunk 2 above)
Comment (diptanu) [inline command/agent/config_test.go]: Thanks!

**Comment by nak3** [comment]
@diptanu Thank you. I updated.

**Pull request merged**